src/com/jalasoft/search/model/Search.java

```diff
@@ -0,0 +1,222 @@
+                                filterResults.add(inputFiles.get(i));
+                            }
+                        }
+                        reader.close();
+                    } catch (FileNotFoundException e) {e.printStackTrace();
+                    } catch (IOException e) {e.printStackTrace();}
+                }
+            }
+            return(filterResults);
+        }
+
+    /**
+     *searchByExtension method searches into a List of Files specific ones that have an specific extension and return List of Files
+     * @param inputFiles
+     * receives a list of Files
+     * @param extension
+     * receives a search criteria
+     * @return
+     * List of Files filtered by Extension criteria
+     */
+    private List <File> searchByExtension(List <File> inputFiles, String extension){
+        List <File> filterResults = new ArrayList<>();
+        if(inputFiles.size()>= 0){
+            for (int i = 0; i< inputFiles.size(); i++){
+                if(inputFiles.get(i).getName().endsWith(extension)){
+                    filterResults.add(inputFiles.get(i));
+                }
+            }
+        }
+        return (filterResults);
+    }
+
+    /**
+     *searchBySize method searches into a List of Files specific ones that have an specific size and return List of Files
+     * @param inputFiles
+     * receives a list of Files
+     * @param size
+     * receives a search criteria
+     * @return
+     * List of Files filtered by Size criteria
+     */
+    private List <File>  searchBySize(List <File> inputFiles, long size){
+        List <File> filterResults = new ArrayList<>();
+        if(inputFiles.size() >=0 ){
+            for (int i = 0; i< inputFiles.size(); i++){
+                if(inputFiles.get(i).length() == size){
+                    filterResults.add(inputFiles.get(i));
+                }
+            }
+        }
+        return (filterResults);
+    }
+
+    /**
+     *searchByModificationDate method searches into a List of Files specific ones that have an modification date and return List of Files
+     * @param inputFiles
+     * receives a list of Files
+     * @param modification
+     * receives a search criteria
+     * @return
+     * List of Files filtered by modification date criteria
+     */
+    private List <File> searchByModificationDate(List <File> inputFiles, long modification){
+        List <File> filterResults = new ArrayList<>();
+        if(inputFiles.size()>= 0){
+            for (int i = 0; i< inputFiles.size(); i++){
+                if(inputFiles.get(i).lastModified() == modification) {
+                    filterResults.add(inputFiles.get(i));
+                }
+            }
+        }
+        return (filterResults);
+    }
+
+    private void fillFile (List <File> inputFiles){
+        if(inputFiles.size()>= 0){
+            for (int i = 0; i< inputFiles.size(); i++){
+                com.jalasoft.search.model.File addFile = new com.jalasoft.search.model.File();
```

Comment:
It will better import the class File instead of access from code

Reply:
Changing to FileSearch.